### ISSUE TITLE
Error submitting Entry to Salesforce with Agree field

### ISSUE THREAD

**Issue by martinherweg** [verbb/formie#1173]
### Describe the bug

We have a a boolean field in Salesforce which is mapped to an Agree Field in formie with the values `true` and `false`, when trying to submit the submission to salesforce as a Lead we get a success message but nothing appears in Salesforce.

When debugging locally I get back this message from Salesforce API

```
[{"message":"Cannot deserialize instance of boolean from VALUE_STRING value 1 or request may be missing a required field (truncated...)
```

when it's unchecked it works fine, it seems like values 1 or 0 are not properly sent as booleans as values which are statically mapped as true and false are. 

I tried to solve myself to be able to submit a PR but then thought there is too much going which I'm not aware of :)

### Steps to reproduce

.

### Form settings

.

### Craft CMS version

4.3.1

### Plugin version

2.0.14

**Comment by engram-design** [comment]
Fixed for the next release. To get this early, change your `verbb/formie` requirement in `composer.json` to:

```
"require": {
  "verbb/formie": "dev-craft-4 as 2.0.17",
  "...": "..."
}
```

Then run `composer update`.

**Comment by engram-design** [comment]
Fixed in [2.0.18](https://github.com/verbb/formie/releases/tag/2.0.18)
